tests/test_requests.py

```diff
@@ -385,6 +385,27 @@ def test_param_cookiejar_works(self, httpbin):
         # Make sure the cookie was sent
         assert r.json()['cookies']['foo'] == 'bar'
 
+    def test_cookies_multiple_domains_returns_correct_value(self, httpbin):
+        """Tests that multiple_domains() returns the correct value. 
+
+        See GH #4557
+        """
+        s = requests.session()
+        r = s.get(httpbin('cookies'))
+        r.cookies.set('foo', 'bar', domain='.google.com')
+
+        assert r.cookies.multiple_domains() == False
+
+        r.cookies.set('baz', 'foo')
+
+        assert r.cookies.multiple_domains() == False
+
+        r.cookies.set('baz', 'foo', domain='.google.com')
```

Comment:
If we’re going to add this to the test let’s verify it’s working as expected. 

We can probably collapse a few of the new lines too if you don’t think it affects readability. Perhaps no extra newline between the cookies.set and assert lines?

Reply:
Is it worth a new test method to make testing the ['.google.com',None,'.google.com'] case a bit less awkward?

Reply:
If you think that makes it easier to grock, I don’t have an issue with it. I’d encourage a glance at pytest’s `parameterize` decorator which will allow you to run the same test with different input.